### ISSUE TITLE
Xamarin Forms Android, return to previous App.MainPage 

### ISSUE THREAD

**Issue by MaxenceSAUNIER** [xamarin/Xamarin.Forms#5236]
### Steps to Reproduce

1. Open my project
2. Click on the button
3. Click on previous button
4. Click on the Android opened program list and resume the app.
5. You are on the first page, not the second and the button not working

### Expected Behavior
Previous button close the app. Resume, open the app on the second page.

### Actual Behavior
Previous button close the app. Resume, open the app on the first page. If you spy on the App.MainPage, have SecondPage in the value but the view show the first page.

### Basic Information

- Version with issue: 4.0.0 pre 4
- Last known good version: don't know
- IDE: Visual Studio
- Platform Target Frameworks: 
  - iOS:  not test
  - Android:  
  - UWP:  not test
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Reproduction Link
[MainPageBackError.zip](https://github.com/xamarin/Xamarin.Forms/files/2860813/MainPageBackError.zip)

**Comment by pauldipietro** [comment]
By "previous" button, you mean the software back button? I did that and reopened the app, which went to the first page, but the button appeared after a few seconds and ended up working. Do you experience different behavior on different Android versions?

**Comment by Moltroon** [comment]
**By "previous" button, you mean the software back button?**
![backbutton_used](https://user-images.githubusercontent.com/22298309/53181134-4312a380-35f7-11e9-88c4-37190d2bcadf.jpg)

**I did that and reopened the app, which went to the first page, but the button appeared after a few seconds and ended up working. Do you experience different behavior on different Android versions?**
I use an ASUS tablet with Android 7.0

I illustrate they manipulations:

![screenshot_20190221-163349](https://user-images.githubusercontent.com/22298309/53181240-7e14d700-35f7-11e9-810f-3cf60cac77d0.jpg)
**I click on the button**

![screenshot_20190221-163354](https://user-images.githubusercontent.com/22298309/53181272-8a009900-35f7-11e9-8d0c-c39d02b3e3af.jpg)
**I click on the previous button**

![screenshot_20190221-163359](https://user-images.githubusercontent.com/22298309/53181304-9c7ad280-35f7-11e9-8925-69f2516ab5ed.jpg)
**I click on the openned app button**

![screenshot_20190221-163408](https://user-images.githubusercontent.com/22298309/53181353-af8da280-35f7-11e9-8453-ffe40afd8987.jpg)
**I click on the app**

![screenshot_20190221-163411](https://user-images.githubusercontent.com/22298309/53181408-c2a07280-35f7-11e9-88d9-5a584d7f9b41.jpg)
I am returned on the first page and the button do not working.
If I add a spy on the App.MainPage, the spy return the second page element.


Normally, I should be on the second page, but I'm not there.

**Comment by monolara** [comment]
I was going to open another issue but this seems like same problem as mine. 

Problem is only on Android, not on iOS.
In my case I set MainPage on App.cs -> OnSleep method. When I send my application to background with debug i can see that MainPage has changed. But when I re-open my app it opens my old mainpage and when i check what is the current main page, i see it is my new main page (with no error).

I faced this issue on version 4.1.0.581479 of Xamarin.Forms. 
I upgraded it to 4.2.0.608146-pre1 and bug was still there.
I downgraded it to 4.0.0.540366 and issue resolved.

I will continue to use 4.0.0.540366 until this bug is resolved, I hope it will be soon because I like to use latest versions ^^
